AuthenticationDemo/ExampleSessionService.swift

```diff
@@ -67,14 +69,45 @@ public final class ExampleSessionService: SessionServiceType {
         })
     }
     
-    private func failureSignalProducer(dispatchTime: dispatch_time_t) -> SignalProducer<ExampleUser, SessionServiceError> {
+    private func logInFailureSignalProducer(dispatchTime: dispatch_time_t) -> SignalProducer<ExampleUser, SessionServiceError> {
         return SignalProducer<ExampleUser, SessionServiceError> { observer, disposable in
             dispatch_after(dispatchTime, dispatch_get_main_queue()) {
-                observer.sendFailed(.InvalidCredentials(.None))
+                observer.sendFailed(.InvalidLogInCredentials(.None))
             }
         }.on(failed: { [unowned self] _ in
-            self._eventsObserver.sendNext(.LogInError(.InvalidCredentials(.None)))
+            self._eventsObserver.sendNext(.LogInError(.InvalidLogInCredentials(.None)))
         })
     }
     
+    public func signUp(name: String, email: Email, password: String) -> SignalProducer<ExampleUser, SessionServiceError> {
+        let dispatchTime = dispatch_time(DISPATCH_TIME_NOW, (Int64)(2 * NSEC_PER_SEC))
+        if email.raw == _email {
+            if _registeredAlready {
+                return signUpFailureSignalProducer(dispatchTime)
+            } else {
+                let user = ExampleUser(email: email.raw, password: password)
+                return signUpSuccessSignalProducer(user, dispatchTime: dispatchTime)
+            }
+        } else {
+            return signUpFailureSignalProducer(dispatchTime)
+        }
+    }
+    
+    private func signUpSuccessSignalProducer(user: ExampleUser, dispatchTime: dispatch_time_t) -> SignalProducer<ExampleUser, SessionServiceError> {
+        return SignalProducer<ExampleUser, SessionServiceError> { observer, disposable in
+            dispatch_after(dispatchTime, dispatch_get_main_queue()) {
+                observer.sendNext(user)
+                observer.sendCompleted()
+            }
+        }
+    }
+    
+    private func signUpFailureSignalProducer(dispatchTime: dispatch_time_t) -> SignalProducer<ExampleUser, SessionServiceError> {
```

Comment:
I would not name this with `SignalProducer`, just `signUpFailure(...) -> ....`
You could leave it like this though if you prefer it this way